core/src/main/scala/kafka/admin/ConsumerGroupCommand.scala

```diff
@@ -71,64 +72,122 @@ object ConsumerGroupCommand {
+    def describe(): Option[Array[PartitionAssignmentState]] = {
+      collectGroupAssignment(opts.options.valueOf(opts.groupOpt)) match {
+        case None =>
+          consumerGroupOutputWriter.printError(s"The consumer group '${opts.groupOpt}' does not exist")
+          None
+        case Some(results) =>
+          consumerGroupOutputWriter.printAssignment(results)
+          Some(results)
+      }
     }
 
     def close(): Unit
 
-    protected def opts: ConsumerGroupCommandOptions
-
-    protected def getLogEndOffset(topic: String, partition: Int): LogEndOffsetResult
+    protected def consumerGroupOutputWriter: OutputWriter
 
-    protected def describeGroup(group: String): Unit
+    protected def opts: ConsumerGroupCommandOptions
 
-    protected def describeTopicPartition(group: String,
-                                         topicPartitions: Seq[TopicAndPartition],
-                                         getPartitionOffset: TopicAndPartition => Option[Long],
-                                         getOwner: TopicAndPartition => Option[String]): Unit = {
-      topicPartitions
-        .sortBy { case topicPartition => topicPartition.partition }
-        .foreach { topicPartition =>
-          describePartition(group, topicPartition.topic, topicPartition.partition, getPartitionOffset(topicPartition),
-            getOwner(topicPartition))
-        }
+    protected def getLogEndOffset(topicAndPartition: TopicAndPartition): LogEndOffsetResult
+
+    protected def collectGroupAssignment(group: String): Option[Array[PartitionAssignmentState]]
+
+    protected def collectMemberAssignment(group: String,
+                                          topicPartitions: Seq[TopicAndPartition],
+                                          getPartitionOffset: TopicAndPartition => Option[Long],
+                                          memberIdOpt: Option[String],
+                                          memberHostOpt: Option[String],
+                                          clientIdOpt: Option[String]): Array[PartitionAssignmentState] = {
+      if (topicPartitions.isEmpty)
+        Array[PartitionAssignmentState](getDescribeResult(group, None, None, None, memberIdOpt, memberHostOpt, clientIdOpt, None))
+      else {
+        var assignmentRows: Array[PartitionAssignmentState] = Array()
+        topicPartitions
+          .sortBy { case topicPartition => topicPartition.partition }
+          .foreach { topicPartition =>
+            assignmentRows = assignmentRows :+ describePartition(group, topicPartition.topic, topicPartition.partition, getPartitionOffset(topicPartition),
+              memberIdOpt, memberHostOpt, clientIdOpt)
+          }
+        assignmentRows
+      }
     }
 
-    protected def printDescribeHeader() {
-      println("%-30s %-30s %-10s %-15s %-15s %-15s %s".format("GROUP", "TOPIC", "PARTITION", "CURRENT-OFFSET", "LOG-END-OFFSET", "LAG", "OWNER"))
+    protected def getDescribeResult(group: String,
```

Comment:
Seems like the only thing this method is contributing is the computation of lag. Maybe we could replace it with a method like this:
```scala
protected def getLag(offset: Option[Long], logEndOffset: Option[Long])
```
What do you think?